Add separate configuration section to documentation

christophfink opened the pull request:
Resolves #90

HTenkanen reviewed (commented):
@christophfink Should we enforce the use of Markdown instead of rst? I think most people are generally more comfortable with Markdown files and Notebooks, hence, I think it might be good to write markdown files when possible. What do you think? 

This would mean that:
- installation.rst --> installation.md
- configuration.rst --> configuration.md

Others would stay as they are.

christophfink commented:
I totally would support that - with rst, I have to look up things, all the time

christophfink commented:
- Converted *installation* and *configuration* to markdown. 
- Added a paragraph to the contribution guidelines
- Fixed a few minor things in the documentation on the way

Please check for factual correctness, but also for language issues (feel a bit tired, words are coming slowly)

![why-waste-time-when-few-word-do-trick](https://user-images.githubusercontent.com/4998473/168677409-ebc8fe1d-d255-492e-94c8-1259cd781675.gif)

wklumpen reviewed (approved):
I only see minor changes (1 line in each file) but they look good!

HTenkanen commented on docs/conf.py:
```diff
@@ -32,10 +32,13 @@
     "sphinx.ext.napoleon",
     "IPython.sphinxext.ipython_console_highlighting",
     #"myst_nb",
+    "myst_parser",
```
Do not add `myst_parser` to conf.py because it causes error in Sphinx build, see commits under this comment:
https://github.com/r5py/r5py/pull/38#pullrequestreview-969167018

HTenkanen replied:
It's weird because Sphinx documentation tells to do this, but apparently these are taken care of by `sphinx_book_theme`.

HTenkanen replied:
Hmm, I take these comments back. It seems that `myst_nb` was taking care of these, but now as we are not using it, the `myst_parser` and `source_suffix` seem to be required.

HTenkanen commented on docs/conf.py:
```diff
@@ -32,10 +32,13 @@
 
 ]
 
+source_suffix = [".rst", ".md"]
```
Do not add `source_suffix` to conf.py because it causes error in Sphinx build, see commits under this comment:
https://github.com/r5py/r5py/pull/38#pullrequestreview-969167018

HTenkanen replied:
My bad. We do seem to need this because `myst_nb` is not used.

HTenkanen commented on docs/index.rst:
```diff
@@ -9,8 +9,9 @@ R5py
+   configuration
    basic-usage.ipynb
-   CONTRIBUTING.md
+   CONTRIBUTING
```
Are you sure this works without the `.md` file extension? Could well be. Mostly asking to update my knowledge if this is the case.

HTenkanen reviewed (changes requested):
Looks good! Couple of questions and one change request for the `conf.py` to avoid Sphinx build errors in RTD.

HTenkanen reviewed (approved):
Everything looks good!
